content/articles/leveraging-openai-gym-and-the-anytrading-environment-for-trading/index.md

```diff
@@ -0,0 +1,233 @@
+If we look at the actions we can take using `environment.action_space`, we'll notice that we only have two actions we can take. We can only `Short` or `Long`. In other algorithms, you can `Hold`. But, this algorithm is only limited to two actions. If you're not familiar with these crucial terminologies, read this [documentation](https://en.wikipedia.org/wiki/Position_(finance)).
+
+```python
+state = env.reset()
+while True: 
+    action = env.action_space.sample()
+    n_state, reward, done, info = env.step(action)
+    if done: 
+        print("info", info)
+        break
+```
+If you've worked with OpenAI Gym before, this first part of the code must be familiar to you. It is the typical code you write when building your test environment.
+        
+The code below visualizes this environment using `matplotlib`.
+
+```python        
+plt.figure(figsize=(20,10))
+plt.cla()
+env.render_all()
+plt.show()
+```
+
+![Random step visualization](/engineering-education//leveraging-openai-gym-and-the-anytrading-environment-for-trading/random-step-visualization.png)
+
+To summarize this section, we are taking a bunch of random steps in our environment and visualizing it.
+
+Now, we can start building our RL agent to try and trade profitably in this environment. 
+
+### Training an RL agent to trade using the Gym environment
+We begin by wrapping our environment inside the dummy vectorized environment wrapper, `DummyVecEnv`.
+
+```python
+env_build = lambda: gym.make('stocks-v0', df=df, frame_bound=(5,30), window_size=5)
+env = DummyVecEnv([env_build])
+```
+We are creating an `env_build` function. We are taking that function and putting it inside the `DummyVecEnv`. Finally, we save the result inside the `env` variable so that when we start building our training model.  We'll now use the `env` variable.
+
+Next, let's set up our algorithm and kick off our training.
+
+```python
+model_train = A2C('MlpLstmPolicy', env, verbose=1) 
+model_train.learn(total_timesteps=100000)
+```
+With these two lines of code, our model will start training.
+
+`A2C` is the algorithm that we will use for this run. `A2C` is an acronym that stands for `Advantage Actor-Critic. It is an RL algorithm that combines Policy-Based and Value-Based RL techniques. You can read more about it [here](https://towardsdatascience.com/advantage-actor-critic-tutorial-mina2c-7a3249962fc8).
+We are using the `MlpLstmPolicy` which is a deep neural network policy with an LSTM layer. It is an important policy as it allows a neural network to keep context and learn about the previous history within its neurons.
+The last line of code kicks off training with `100000` timesteps. Ideally, what you need to observe while training is the `explained_variance` value. You want it to be as high as possible. We are looking at values between `0` and `1`. A number close to `1` denotes a high value while that close to `0`, a low value. You should also make sure the `value_loss` is as low as possible.
```

Comment:
Revise the use of backticks in these two paragraphs to highlight the correct words